opti/problem.py

```diff
@@ -34,6 +38,7 @@ def __init__(
         data: Optional[DataFrameLike] = None,
         optima: Optional[DataFrameLike] = None,
         name: Optional[str] = None,
+        equalities: Optional[List[str]] = None,
```

Comment:
`Problem` should not be modified.
All constraints are already in Problem.constraints.

Reply:
Okay, but if we want to recover the solution (i.e. outputs _and_ inputs) of the initial, unreduced problem, we need to be able to backtransform from the lower dimensional input space to the affine space defined by the original linear equality constraints. Since we "forget" the equality constraints when we assemble the reduced problem (no linear equality constraints present in case of the reduced problem), we need to save the information about the backtransformation somewhere else, right? At least if we are not only interested in the outputs but also in the position of the population in the input space... 
If you agree: As an alternative to changing `Problem` we could add the correct backtransformation to the returns of `reduce()`, would this be better?